### PR TITLE
enable nullable reference types

### PR DESCRIPTION
**Pull request by eyalzus12**
looking at the source code from calamity, the last argument of `BossRushEntry` (aka `needsDead`) is nullable and the rest aren't.
from documentation, it seems that passing that argument as null would default it to the boss, so perhaps there's no need in us defaulting to the boss manually.

I made VesperaEnchStone non-nullable since the code that uses it should run after `Load` (i think) and never after `Unload`. So the `null!` should be safe here.

### CODE REVIEW

File: Common/Systems/BossRushInjectionSystem.cs [hunk 1]
```diff
@@ -23,7 +23,7 @@ namespace TheBereftSouls.Common.Systems;
 //
 // Referenced permalink:
 // - https://github.com/CalamityTeam/CalamityModPublic/blob/a5bdc9231f2859abddb85c9043413a57fcb042b9/ModSupport/ModCalls.cs#L1999
-using BossRushEntry = (int, int, Action<int>, int, bool, float, int[], int[]);
+using BossRushEntry = (int, int, Action<int>, int, bool, float, int[], int[]?);
```
Comment (1Git2Clone) [inline Common/Systems/BossRushInjectionSystem.cs:26]: Why is this last entry nullable? In the `AddToBossRush` function we do default it to `[NPCType]`.

Comment (eyalzus12) [reply Common/Systems/BossRushInjectionSystem.cs:26]: written in the pr description. the calamity source code may put null in the last entry, which seems to be a default to `[NPCType]` (handled inside calamity).

**Comment by 1Git2Clone** [comment]
I find it better to explicitly have certain parts be nullable when needed rather than allowing it on a project level since that'd suppress all the nullable warnings<sup>\[[1]\]</sup>, making the project harder to debug in cases of null related errors.

[1]: https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/nullable-warnings

**Comment by eyalzus12** [comment]
> I find it better to explicitly have certain parts be nullable when needed rather than allowing it on a project level since that'd suppress all the nullable warnings[[1](https://learn.microsoft.com/en-us/dotnet/csharp/language-reference/compiler-messages/nullable-warnings)], making the project harder to debug in cases of null related errors.


nullable warnings are only enabled when nullable reference types are enabled...

**Pull request merged**